raritan/decorators.py

```diff
@@ -173,30 +183,37 @@ def wrapper_function(*args, **kwargs):
                         # It is not optional
                         if not inner_optional_flag:
                             raise FileNotFoundError(f"Non-Optional file missing: {name}")
+                        # It is optional, using a dictionary provided to make an empty dataframe with column names.
                         else:
-                            logger.info(f"Optional file missing: {name}")
-                            continue  # Skip processing if file is optional
-                    # Pass them on to the input handler.
-                    duration, data = _time_function(settings.input_handler, *[group, name])
-                    if filters is not None:
-                        try:
-                            for filter_function, value in filters.items():
-                                data = filter_function(data, value)
-                        except Exception as e:
-                            error(f"Something went wrong with the filter function: {e}")  # Log the error message
-                            # We want all the flows to run even if one fails.
-                            # After the build is complete we scan the output for `Traceback` and if the key word is found,
-                            # it will throw a fail on Jenkins.
-                            quit()
-                    context.set_data_reference(key, data)
-                    message = ''
-                    # Allow an analyze_asset_handler to ensure integrity and/or write the logging.
-                    if analyze and hasattr(settings, 'analyze_asset_handler'):
-                        message = settings.analyze_asset_handler(group, name, None, data, duration, 'input')
-                    if message is None or len(message) == 0:
-                        message = f'Loaded asset: {name} {duration}'
-                    logger.success(message)
+                            logger.info(f"Optional file missing: {name}, using default dictionary.")
+                            if default_dictionary:
+                                data = pd.DataFrame(columns=default_dictionary.keys()).astype(
```

Comment:
Could we pass all of this off to the input_handler? I think the values in "name" like "default_dictionary" are already going there. Otherwise, we'll have to include pandas in our packages, which I'd like to continue avoiding.

Reply:
Removed pandas and moved that work to the ETL repo

Reply:
Woohoo! Thanks Jack!